### PR TITLE
update button style, wrap buttons inside container

### PR DESCRIPTION
**Pull request by wouterlanduydt**
I'm working on a project that has a lot of themes and thought of a small tweak that would make it easier to choose the correct theme.

In this PR I've added the ability to add a `themeColor` that is represent in the button (see screenshot) & made the buttons wrap in the addon container.

Screenshot 2020-02-15 at 18 13 23

### CODE REVIEW

**Comment by icew1nd** [comment]
Love this! 🔥 

Was actually just about to add the flex-wrap: wrap my self.

Maybe themeColor should be named color - like name isn't called themeName as input. Maybe the example should be updated as well?

**Comment by wouterlanduydt** [comment]
> Maybe themeColor should be named color

That's a good suggestion.

I also nested it in an `addonOptions` property because a general property name like `color` could easily conflict with an actual theme property. So this is an example on how to pass themes with colors for the addon:
```
const defaultTheme = {
  name: "Default",
  addonOptions: {
    color: "#306e21",
  },
 [all theme props]
};
```
In my opinion it would be cleaner & provide more flexibility to future development of the plugin to pass theme like this, but that would be a breaking change:
```
const defaultTheme = {
  name: "Default",
  color: "#306e21",
  theme: [all theme props]
};
```
@echoulen Thoughts on that?

File: src/Themes.tsx [hunk 1]
```diff
@@ -74,37 +76,46 @@ export const Themes = compose<BaseComponentProps, ThemeProps>(
...
-    padding: 13px;
-    margin-right: 15px;
-    height: 55px;
+const Button = styled.button<ButtonProps>`
+    display: flex;
+    align-items: center;
+    border: none;
+    margin: 0 0;
+    margin-right: 8px;
+    margin-bottom: 8px;
```
Comment (echoulen) [inline src/Themes.tsx]: can be merged to `margin: 0 8px 8px 0`

File: src/Themes.tsx [hunk 2]
```diff
@@ -74,37 +76,46 @@ export const Themes = compose<BaseComponentProps, ThemeProps>(
...
+    padding: 0;
+    padding-right: 8px;
+    padding-left: ${props => props.themeColor ? 0 : "8px"};
```
Comment (echoulen) [inline src/Themes.tsx]: merge to `padding: 0 8px 0 ${props => props.themeColor ? 0 : "8px"}`

File: src/Themes.tsx [hunk 3]
```diff
@@ -25,19 +25,21 @@ interface ThemeHandler {
...
-        <FillingDiv />
-        <Border>|</Border>
-    </FlexRow>
+    <div>
+        <ButtonContainer>
+            {themes.map((th, i) => (
+                <Button
+                    id={`theme-selection-button-${th.name}`}
+                    selected={th === theme}
+                    themeColor={th.addonOptions?.themeColor}
```
Comment (echoulen) [inline src/Themes.tsx]: I'm not sure `?` will be broken when `this.addonOptions === undefined` or not
my you try this case with `undefined` addonOptions?

**Pull request closed without merging**